src/observable.ts

```diff
@@ -1,58 +1,117 @@
-export type ObservableConstructor<T> = new (subscribe: (observer: IObserver<T>) => Unsubscribe) => IObservable<T>;
+import "symbol-observable";
 
+// Based on https://github.com/tc39/proposal-observable
 export interface IObservable<T> {
-    subscribe(next: (value: T) => void): Unsubscribe;
-    filter(condition: (value: T) => boolean): IObservable<T>;
-    map<U>(select: (value: T) => U): IObservable<U>;
-    resolved<P>(this: IObservable<Promise<P>>): IObservable<P>;
-    rejected(this: IObservable<Promise<any>>): IObservable<any>;
-    asObservable(): IObservable<T>;
+    subscribe(observer?: IObserver<T>): ISubscription;
+    subscribe(next?: (value: T) => void, error?: (error: any) => void, complete?: () => void): ISubscription;
+    subscribe(nextOrObserver?: IObserver<T> | ((value: T) => void), error?: (error: any) => void, complete?: () => void): ISubscription;
+}
+
+export interface ISimpleObservable<T> extends IObservable<T> {
+    filter(condition: (value: T) => boolean): ISimpleObservable<T>;
+    map<U>(select: (value: T) => U): ISimpleObservable<U>;
+    resolved<P>(this: ISimpleObservable<Promise<P>>): ISimpleObservable<P>;
+    rejected(this: ISimpleObservable<Promise<any>>): ISimpleObservable<any>;
+    asObservable(): ISimpleObservable<T>;
+    merge<O>(this: ISimpleObservable<IObservable<O>>): ISimpleObservable<O>;
+    errored(this: ISimpleObservable<IObservable<any>>): ISimpleObservable<any>;
+    completed(this: ISimpleObservable<IObservable<any>>): ISimpleObservable<void>;
+    [Symbol.observable](): IObservable<T>; // uses symbol-observable ponyfill
+}
+
+export interface ISubscription {
+    unsubscribe(): void;
+    closed: boolean;
 }
 
 export type Unsubscribe = () => void;
 
 export interface IObserver<T> {
-    next(value: T): void;
+    next?(value: T): void;
+    error?(err: any): void;
+    complete?(): void;
+    closed?: boolean;
 }
 
-class SimpleObservable<T> implements IObservable<T> {
-    constructor(private _subscribe: (observer: IObserver<T>) => Unsubscribe) { }
+export type SubscriberFunction<T> = (observer: IObserver<T>) => ISubscription;
+
+class SimpleObservable<T> implements ISimpleObservable<T> {
+    constructor(private _subscribe: SubscriberFunction<T>) { }
+
+    [Symbol.observable](): IObservable<T> { return this };
+
+    subscribe(observer?: IObserver<T>): ISubscription;
+    subscribe(next?: (value: T) => void, error?: (error: any) => void, complete?: () => void): ISubscription;
+    subscribe(nextOrObserver?: IObserver<T> | ((value: T) => void), error?: (error: any) => void, complete?: () => void): ISubscription {
+        return this.isObserver(nextOrObserver)

```

Comment:
I think we need an Observer class to wrap these functions/IObserver so we don't have to null-check the IObserver methods everywhere, and the wrapAsync function doesn't need to re-implement this isObserver logic (it's actually missing this check at the moment).